package/sbp_j1939_bridge/src/sbp_j1939_bridge.c

```diff
@@ -26,10 +27,108 @@
 #define J1939_SUB_ENDPOINT "ipc:///var/run/sockets/j1939_internal.pub" /* J1939 Internal Out */
 #define J1939_PUB_ENDPOINT "ipc:///var/run/sockets/j1939_internal.sub" /* J1939 Internal In */
 
+#define J1939_HEADER_LENGTH 4
+
 bool j1939_debug = false;
 
+struct imu_raw_msg {
+  u32 tow;
+  u8  tow_f;
+  u16 acc_x;
+  u16 acc_y;
+  u16 acc_z;
+  u16 gyr_x;
+  u16 gyr_y;
+  u16 gyr_z;
+};
+
 struct j1939_sbp_state {
-  int counter;
+  u32 PGN;
+  u8 priority;
+  bool extended_data_page;
+  bool data_page;
+  u8 pdu_format;
+  u8 pdu_specific;
+  u8 source_address;
+  u8 *payload;
+  u8 payload_length;
+  struct imu_raw_msg imu_raw;
+};
+
+struct j1939_message {
+  int (*handler)(const u8 *payload, const u8 payload_length, struct j1939_sbp_state *state);
+  u32 PGN;
+  u8 payload_length;
+};
+
+static int handle_slope_sensor_info_2(const u8 *payload, const u8 payload_length, struct j1939_sbp_state *state)
+{
+  u32 pitch = 0;
+  for (int i = 0; i < 3; i++) {
+    pitch += payload[i] << (8 * i);
+  }
+
+  u32 roll = 0;
+  for (int i = 0; i < 3; i++) {
+    roll += payload[i + 3] << (8 * i);
+  }
+
+  (void) payload_length;
+  (void) state;
+  piksi_log(LOG_ERR, "slope - pitch: %.2f roll %.2f", pitch / 32768.0 - 250, roll / 32768.0 - 250);
+
+  return 0;
+}
+
+static int handle_angular_rate(const u8 *payload, const u8 payload_length, struct j1939_sbp_state *state)
+{
+  u16 angular_velocity[3] = {0, 0, 0}; /* x, y, z axes */
+  for (int i = 0; i < 3; i++) {
+    angular_velocity[i] = payload[i*2] + (payload[i*2 + 1] << 8);
+  }
+
+  (void) payload_length;
+  piksi_log(LOG_ERR, "Angular Rate X, Y, Z : %.2f, %.2f, %.2f", angular_velocity[0] / 128.0 - 250, angular_velocity[1] / 128.0 - 250, angular_velocity[2] / 128.0 - 250);
+
+  state->imu_raw.gyr_x = angular_velocity[0];
+  state->imu_raw.gyr_x = angular_velocity[1];
+  state->imu_raw.gyr_x = angular_velocity[2];
+
+  return 0;
+}
+
+static int handle_acceleration_sensor(const u8 *payload, const u8 payload_length, struct j1939_sbp_state *state)
+{
+  u16 acceleration[3] = {0, 0, 0}; /* x, y, z axes */
+  for (int i = 0; i < 3; i++) {
+    acceleration[i] = payload[i*2] + (payload[i*2 + 1] << 8);
+  }
+
+  (void) payload_length;
+  piksi_log(LOG_ERR, "Acceleration X, Y, Z : %.2f, %.2f, %.2f", (acceleration[0] - 32000) / 100.0, (acceleration[1] - 32000) / 100.0, (acceleration[2] - 32000) / 100.0);
+
+  state->imu_raw.acc_x = acceleration[0];
+  state->imu_raw.acc_x = acceleration[0];
```

Comment:
This looks like a copy-paste error too, should probably be [0], [1], [2] and acc_x, acc_y, acc_z?